contribute/design/openebs-provisioner-high-availibility.md

```diff
@@ -0,0 +1,77 @@
+# High Availability for OpenEBS Volume Provisioner
+
+## Problems addressed:
+
+- `openebs-provisioner` should always be available.
+- `openebs-provisioner` should always be running on the specified/labeled nodes. 
+- If any node in cluster goes down in that scenario the pod should be assigned to other node in the cluster.
+ 
+## Solutions for above problems: 
+
+
+- Increase the `replicas` to 2 or >2 in `openebs-operator.yaml` for `openebs-provisioner`. 
+
+
+- We need to make sure that each node is running at least one  `openebs-provisioner` pod.  
+  We need to label nodes to assign the pods on those. 
+
+  To do that execute: 
+
+```
+kubectl label nodes kubeminion-01  node=minion01  # For node kubeminion-01
```

Comment:
It will be ideal to avoid modifications to the K8s setup. This should be the last resort.